app/addons/components/components/deletedatabasemodal.js

```diff
@@ -64,7 +65,7 @@ export class DeleteDatabaseModal extends React.Component {
   onDeleteClick = (e) => {
     e.preventDefault();
 
-    Actions.deleteDatabase(this.getDatabaseName());
+    Actions.deleteDatabase(this.getDatabaseName(), this.props.onSuccess);
```

Comment:
I'm kind of torn between rather just returning a promise from the `Actions.deleteDatabase` and then calling `this.props.onSuccess`. What do you think?